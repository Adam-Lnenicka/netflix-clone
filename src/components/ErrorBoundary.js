import React from "react";

function ErrorBoundary(props) {
  const ErrorMessage = () => {
    return (
      <div>
        <p>Sorry, cannot display content at the moment.</p>
      </div>
    );
  };

  return <>{props.children ? props.children : <ErrorMessage />}</>;
}
export default ErrorBoundary;
