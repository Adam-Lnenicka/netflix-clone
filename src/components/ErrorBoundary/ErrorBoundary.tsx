import React, { Component } from "react";
import ErrorBoundaryComponent from "./component";

interface State {
  hasError: boolean;
}
class ErrorBoundary extends Component<State> {
  state: State = { hasError: false };

  componentDidCatch(error, errorInfo) {
    this.setState({
      hasError: true,
      errorName: error.toString(),
      componentStack: errorInfo.componentStack,
    });
  }
  render() {
    const { errorName, componentStack, hasError } = this.state;
    return hasError ? (
      <ErrorBoundaryComponent
        errorName={errorName}
        componentStack={componentStack}
      />
    ) : (
      this.props.children
    );
  }
}

export default ErrorBoundary;
