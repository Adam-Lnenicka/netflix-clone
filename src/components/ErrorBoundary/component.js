import React from "react";
import { ErrorBoundaryContainer } from "./styles/ErrorBoundaryStyles";

const ErrorBoundaryComponent = ({ errorName, componentStack }) => {
  return (
    <ErrorBoundaryContainer>
      <h2>Sorry, something went wrong!</h2>
      <h3>Here is some info:</h3>
      <div>
        <p>Error Type:</p>
        <span>{errorName}</span>
      </div>
      <article>
        <p>Stacktrace:</p>

        {(componentStack + "")
          .trim()
          .split("\n")
          .map((element, i) => (
            <p key={i}>
              <span>{element}</span>
            </p>
          ))}
      </article>
    </ErrorBoundaryContainer>
  );
};

export default ErrorBoundaryComponent;
