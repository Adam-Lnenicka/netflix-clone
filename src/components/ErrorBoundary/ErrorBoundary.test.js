import React from "react";
import { shallow } from "enzyme";
import ErrorBoundary from "./ErrorBoundary";

describe("ErrorBoundary ", () => {
  it("Message in component", () => {
    const component = shallow(
      <ErrorBoundary errorName="errorName" componentStack="componentStack" />
    );
    component.setState({ hasError: true });
    expect(component).toMatchSnapshot();
  });
});
