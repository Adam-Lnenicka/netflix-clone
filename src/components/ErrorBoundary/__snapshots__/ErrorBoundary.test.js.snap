// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary  Message in component 1`] = `<ErrorBoundaryComponent />`;
